EMGLog.m

```diff
@@ -0,0 +1,17 @@
+#import <Foundation/Foundation.h>
+#import "EMGLog.h"
+
+void EMGLog(NSString *format, ...) {
+    static dispatch_once_t queueCreationGuard;
+    static NSUUID *uuid;
+    dispatch_once(&queueCreationGuard, ^{
+        uuid = [NSUUID new];
+    });
+    
+    va_list args;
+    va_start(args, format);
+    NSString *input = [[NSString alloc] initWithFormat:format arguments:args];
+    va_end(args);
+    NSLog(@"### EMG %@ %@", uuid, input);
+    printf("EMERGE: %s \n", [input cStringUsingEncoding:NSUTF8StringEncoding]);
```

Comment:
I added an extra `printf`, for some reason `NSLog` wasn't reporting to the console logs when I was testing my sample app.